atom/kite/lib/kite.js

```diff
@@ -332,9 +333,93 @@ var KiteIncoming = {
     }
     return null;
   },
-
 };
 
+var CompletionsProvider = {
+  selector: '.source.python',
+  disableForSelector: '.source.python .comment, .source.python .string',
+  inclusionPriority: 2,
+  suggestionPriority: 2,
+  excludeLowerPriority: false,
+
+  // called to handle diff suggestions
+  getSuggestions: function(params) {
+    return new Promise(function (resolve, reject) {
+      var text = params.editor.getText();
+      var cursor = KiteOutgoing.pointToOffset(text, params.bufferPosition);
```

Comment:
if you were to split `CompletionsProvider` into its own module, you'll have to move this function to a shared module which makes sense since `pointToOffset` is just a utility function that operates on native js and atom objects.